app/src/features/Mappings/Edit/EditMapping.tsx

```diff
@@ -0,0 +1,334 @@
+import clsx from "clsx";
+import differenceBy from "lodash/differenceBy";
+import isEqual from "lodash/isEqual";
+import { useTranslation } from "react-i18next";
+import { useHistory, useParams } from "react-router-dom";
+
+import { useAppSelector } from "app/store";
+import useEditMapping from "common/hooks/useEditMapping";
+import { columnSelectors } from "features/Columns/columnSlice";
+import { filterSelectors } from "features/Filters/filterSlice";
+import { joinSelectors } from "features/Joins/joinSlice";
+import TableStep from "features/Mappings/Create/TableStep";
+import { resourceSelectors } from "features/Mappings/resourceSlice";
+import {
+  useApiColumnsCreateMutation,
+  useApiColumnsUpdateMutation,
+  useApiColumnsDestroyMutation,
+  useApiFiltersDestroyMutation,
+  useApiFiltersCreateMutation,
+  useApiFiltersUpdateMutation,
+  useApiJoinsDestroyMutation,
+  useApiJoinsCreateMutation,
+  useApiJoinsUpdateMutation,
+  useApiResourcesUpdateMutation,
+} from "services/api/endpoints";
+import {
+  ColumnRequest,
+  FilterRequest,
+  JoinRequest,
+} from "services/api/generated/api.generated";
+
+const useStyles = makeStyles((theme) => ({
+  contentContainer: {
+    paddingBlock: theme.spacing(4),
+  },
+  button: {
+    textTransform: "none",
+  },
+  previousButton: {
+    margin: theme.spacing(2),
+    color: theme.palette.text.secondary,
+    "&:hover": {
+      backgroundColor: "inherit",
+      color: theme.palette.text.primary,
+    },
+  },
+}));
+
+const EditMapping = (): JSX.Element => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const history = useHistory();
+  const [isEditLoading, setEditLoading] = useState(false);
+  const { sourceId, mappingId } = useParams<{
+    sourceId?: string;
+    mappingId?: string;
+  }>();
+  const { isLoading, data: initialState } = useEditMapping();
+  const { resource, filters, joins, columns } = useAppSelector((state) => ({
+    resource: resourceSelectors.selectById(state, mappingId ?? ""),
+    filters: filterSelectors.selectAll(state),
+    joins: joinSelectors.selectAll(state),
+    columns: columnSelectors.selectAll(state),
+  }));
+  const mapping = useAppSelector((state) =>
+    resourceSelectors.selectById(state, mappingId ?? "")
+  );
+
+  const [updateResource] = useApiResourcesUpdateMutation();
+
+  const [deleteColumn] = useApiColumnsDestroyMutation();
+  const [createColumn] = useApiColumnsCreateMutation();
+  const [updateColumn] = useApiColumnsUpdateMutation();
+
+  const [deleteFilter] = useApiFiltersDestroyMutation();
+  const [createFilter] = useApiFiltersCreateMutation();
+  const [updateFilter] = useApiFiltersUpdateMutation();
+
+  const [deleteJoin] = useApiJoinsDestroyMutation();
+  const [createJoin] = useApiJoinsCreateMutation();
+  const [updateJoin] = useApiJoinsUpdateMutation();
+
+  const handleCancelClick = () => {
+    history.goBack();
+  };
+  const handleEditSubmit = async () => {
```

Comment:
omg I didn't see this one coming lol

Reply:
I agree that this is a gigantic mess and will be hard to maintain but the question is: would it be easier if we sent a request to the API every time the user add/update/delete a column/filter/join ? @tevariou suggested that the JoinSelect, ColumnSelect, FilterSelect handle the API call themselves but the issues are:
- callback hell (passing the cb as props down the entire chain of components)
- these components wouldn't be re-usable with the ones from the stepper
right ?

Reply:
> I don't see the spinner when "save changes" is clicked but I see it in the code ?

It will most likely appear if there are lots of creations

Reply:
I can't say I really like this solution, because rollback/cancels won't be possible and each change will be definitive

Reply:
@simonvadee @BPierrick  Sorry to just jump in the conversation 😅 

UX-wise and for this specific case, I guess I prefer the scenario when all your actions (adding a filter, modifying a join...) have to be saved before it is taken into account. That being said, I know that in some other cases for example when the user wants to change a source column for an attribute mapping, I chose the other scenario in Figma : each change is directly sent to the API, no need to save. 

I'm not shocked if we use both behaviours in the app. It's already like this today in Pyrog.
But I admit that I don't fully realize the frontend consequences for these 2 options, so it's mainly your call at this stage 👍